File: Javascript/Patterns.js
//  * * * *
//  *     *
//  *     *
//  * * * *

// let rows = 4
// let cols = 5

// for (let i = 1; i <= rows; i++) {
//     let row = ""
//     for (let j = 1; j <= cols; j++) {
//         if (i == 1 || i == rows || j == 1 || j == cols) {
//             row += "*"
//         } else {
//             row += " "
//         }
//     }
//     console.log(row);

// }


// * 
// * *
// * * *
// * * * *

// let rows = 4;


// for (let i = 1; i <= rows; i++) {
//     let prints = ""
//     for (let j = 1; j <= i; j++) {
//         prints += "* "
//     }
//     console.log(prints);
// }


// * * * * *
// * * * *
// * * * 
// * *
// *


let rows = 4;

for (let i = 1; i <= rows; i++) {
    let prints = ""
    for (let j = rows; j >= i; j--) {
        prints += "* "
    }
    console.log(prints);
}

File: Python/PractiseSets/Patterns.py
# 1)
# ****
# ****
# ****
# ****

# for i in range(1, 5):
#     for j in range(1,5):
#         print("*", end=" ")
#     print()        



# * * * *
# *     *
# *     *
# * * * *

# rows = 4
# cols = 5

# for i in range(1, rows + 1):
#     for j in range(1, cols + 1):
#         if i == 1 or i == rows or j == 1 or j == cols:
#             print("* ", end="")
#         else:
#             print("  ", end="")
#     print()


# * 
# * * 
# * * * 
# * * * * 

# rows = 4

# for i in range(1, rows + 1):
#     for j in range(1, i + 1):
#         print("* ", end="")
#     print()


# * * * * *
# * * * *
# * * * 
# * *
# *

# rows= 5

# for i in range(1,rows+1):
#     for j in range(rows+1,i,-1):
#         print("* ",end="")
#     print()
    
    
#         * 
#       * * 
#     * * * 
#   * * * * 

rows= 5

for i in range(1,rows+1):
    for j in range(rows+1,i,-1):
        print("  ",end="")
    for k in range(1,i):
        print("* ",end="")
    print()